app/controllers/feedbacks_controller.rb

```diff
@@ -61,6 +62,19 @@ def set_feedback
     authorize @feedback
   end
 
+  def create_redirect
+    if session[:request_url].include?('last_submitted_hours_and_feedbacks')
+      if @feedbackable.class == Assignment
```

Comment:
`@feedback` has the `group_offer?` and `assignment?` methods, that will do what you want a bit slimmer.